packages/core/src/hooks/useContractFunction.ts

```diff
@@ -72,21 +44,21 @@ export function useContractFunction(contract: Contract, functionName: string, op
 
     try {
       transaction = await contractWithSigner[functionName](...args)
-      setState({ transaction, status: 'Mining', chainId })
+      setState({ transaction, status: 'Mining', errored: false, chainId })
       addTransaction({
         transaction,
         submittedAt: Date.now(),
         transactionName: options?.transactionName,
       })
 
       const receipt = await transaction.wait()
-      setState({ receipt, transaction, status: 'Success', chainId })
+      setState({ receipt, transaction, status: 'Success', chainId, errored: false })
     } catch (e) {
       const errorMessage = e.reason ?? e.message
       if (transaction) {
-        setState({ status: 'Fail', transaction, receipt: e.receipt, errorMessage, chainId })
+        setState({ status: 'Fail', transaction, receipt: e.receipt, errorMessage, chainId, errored: true })
       } else {
-        setState({ status: 'Exception', errorMessage, chainId })
+        setState({ status: 'Exception', errorMessage, chainId, errored: true })
```

Comment:
You noticed there's an errorMessage value defined there, but not in other places? Because it's meant to show same thing you proposed {errored:true}